src/libraries/illink-oob.targets

```diff
@@ -0,0 +1,67 @@
+<Project>
+
+  <Target Name="ILLinkTrimOOBAssemblies"
+          AfterTargets="Build"
+          DependsOnTargets="PrepareForAssembliesTrim">
+
+    <Message Text="Trimming $(PackageRID) OOB assemblies with ILLinker..." Importance="high" />
+
+    <PropertyGroup>
+      <LibrariesTrimmedOOBArtifactsPath>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'ILLinkTrimAssembly', '$(BuildSettings)', 'trimmed-oobs'))</LibrariesTrimmedOOBArtifactsPath>
+    </PropertyGroup>
+
+    <ItemGroup>
+
+      <!-- The following is the list of all the OOBs we will ignore for now -->
+      <_OOBsToIgnore Include="System.CodeDom" />
+      <_OOBsToIgnore Include="System.ComponentModel.Composition" />
+      <_OOBsToIgnore Include="System.ComponentModel.Composition.Registration" />
+      <_OOBsToIgnore Include="System.Composition.AttributedModel" />
+      <_OOBsToIgnore Include="System.Composition.Convention" />
+      <_OOBsToIgnore Include="System.Composition.Hosting" />
+      <_OOBsToIgnore Include="System.Composition.Runtime" />
+      <_OOBsToIgnore Include="System.Composition.TypedParts" />
+      <_OOBsToIgnore Include="System.Configuration.ConfigurationManager" />
+      <_OOBsToIgnore Include="System.Speech" />
+
+      <_OOBsToTrim Include="$(NetCoreAppCurrentRuntimePath)*.dll" Exclude="$(NetCoreAppCurrentRuntimePath)*.Generator.dll;$(NetCoreAppCurrentRuntimePath)*.Native.dll" />
+      <_RuntimePackTrimmedAssemblies Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
+
+      <!-- Move previous items to FileName so that we can subtract them -->
+      <_OOBsToFileName Include="@(_OOBsToTrim -> '%(FileName)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </_OOBsToFileName>
+      <_RuntimePackAssembliesToFileName Include="@(_RuntimePackTrimmedAssemblies -> '%(FileName)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </_RuntimePackAssembliesToFileName>
+
+      <_FinalOOBsFileName Include="@(_OOBsToFileName)" Exclude="@(_RuntimePackAssembliesToFileName);@(_OOBsToIgnore)" />
+      <_FinalOOBReferences Include="@(_OOBsToFileName)" Exclude="@(_FinalOOBsFileName)" />
+      <_FinalOOBs Include="@(_FinalOOBsFileName -> '%(OriginalIdentity)')" />
+      <_FinalReferences Include="@(_FinalOOBReferences -> '%(OriginalIdentity)')" />
+      <_FinalReferences Include="$(SystemPrivateCoreLibPath)" />
+
+      <OOBRootAssemblies Include="@(_FinalOOBs)">
+        <RootMode>library</RootMode>
```

Comment:
If the goal is to get as close as possible to the documented experience in https://docs.microsoft.com/en-us/dotnet/core/deploying/prepare-libraries-for-trimming, we shouldn't set RootMode here (though I don't think it makes a difference in practice).

Reply:
That's a good catch, I'll remove it. Turns out doing this actually added some new warnings, and the reason for that is that we don't currently run the linker on our OOBs, so we do have some non-public code that the linker was trimming when using library mode, which was producing additional warnings. I'll fix this and baseline the additional warnings.